rust/frontend/src/impls/service_based_frontend.rs

```diff
@@ -938,6 +950,12 @@ impl ServiceBasedFrontend {
     ) -> Result<DeleteCollectionRecordsResponse, DeleteCollectionRecordsError> {
         let mut records = Vec::new();
 
+        // NOTE(c-gamble): We are using a non-standard metering pattern in which we create a metering context within a
```

Comment:
I think we only need to do latency tracking for 

add()/upsert()/update()

get()/query()